### PR TITLE
Return numerically maximal index (unsigned) or -1 (signed) for no match case 

### PR DESCRIPTION
**Pull request by magehrig**
Address issue https://github.com/ethz-asl/libnabo/issues/2.

### CODE REVIEW

File: nabo/index_heap.h [hunk 1]
```diff
@@ -73,7 +73,12 @@ namespace Nabo
 		typedef typename Eigen::Matrix<Index, Eigen::Dynamic, 1> IndexVector;
 		//! vector of values
 		typedef typename Eigen::Matrix<Value, Eigen::Dynamic, 1> ValueVector;
-		
+
+		//! invalid index
+		static constexpr auto InvalidIndex = std::numeric_limits<IT>::max();
```
Comment (HannesSommer) [inline nabo/index_heap.h]: Once you've converted this to a templated constexpr inline function, you could return as value : 

```c++
std::is_unsigned<IT>::value ? std::numeric_limits<IT>::max() : IT(-1); 
```

File: nabo/nabo.h [hunk 2]
```diff
@@ -213,7 +213,17 @@ namespace Nabo
 	#define NABO_VERSION "1.0.6"
 	//! version of the Nabo library as an int
 	#define NABO_VERSION_INT 10006
-	
+
```
Comment (HannesSommer) [inline nabo/nabo.h]: If you make another commit, could you add a comment here, like:
```c++
// TODO (c++14) Convert invalidIndex, invalidValue to constexpr templated variables
```

**Comment by HannesSommer** [comment]
Aha, I believe you've found a bug in the unit tests:
In
https://github.com/ethz-asl/libnabo/blob/b1ac136d08554bb74b96395f6cc5226b13e5cae9/tests/knnvalidate.cpp#L181
there should have bin `T` instead of `float`.

But now there should probably be `invalidValue<T>()` or the `::InvalidValue` if accessible.

File: tests/knnvalidate.cpp [hunk 3]
```diff
@@ -190,8 +190,8 @@ void validate(const char *fileName, const int K, const int dim, const int method
 				}
 				const Vector pkdtree(d.col(pkdt));
 				const Vector pq(q.col(i));
-				const float distDiff(fabsf((pbf-pq).squaredNorm() - (pkdtree-pq).squaredNorm()));
-				if (distDiff > numeric_limits<float>::epsilon())
+				const T distDiff(fabsf((pbf-pq).squaredNorm() - (pkdtree-pq).squaredNorm()));
+				if (distDiff > numeric_limits<T>::epsilon())
```
Comment (HannesSommer) [inline tests/knnvalidate.cpp]: Good catch :)

**Comment by magehrig** [comment]
I first have to find out why the tests do not pass

**Comment by magehrig** [comment]
Would be nice if you take a quick look at the test. What i have found out so far is that [indexes_kdtree is filled with -1 and dists2_kdtree is filled with 0](https://github.com/ethz-asl/libnabo/blob/e7d295b37d0bd7e0952f10a173dd580ed8703425/tests/knnvalidate.cpp#L163-L165) if `j == 1u`; `j > 1u` succeeds.

**Comment by HannesSommer** [comment]
Okay, I'll do that. Not now but later, probably weekend. Thanks!

**Comment by magehrig** [comment]
@HannesSommer Should be fixed now ;)
